ci/generate-and-upload-build-steps.sh

```diff
@@ -1,15 +1,73 @@
 #!/bin/bash
 set -euo pipefail
 
+upload_build_configuration_step() {
+    cat "ci/gdk_build.template.steps.yaml" | \
+    sed "s|ENGINE_COMMIT_HASH_PLACEHOLDER|$1|g" | \
+    sed "s|BUILD_PLATFORM_PLACEHOLDER|$2|g" | \
+    sed "s|BUILD_TARGET_PLACEHOLDER|$3|g" | \
+    sed "s|BUILD_STATE_PLACEHOLDER|$4|g" | \
+    buildkite-agent pipeline upload
+}
+
+generate_build_configuration_steps () {
+    # See https://docs.unrealengine.com/en-US/Programming/Development/BuildConfigurations/index.html for possible configurations 
+
+    # if BUILD_ALL_CONFIGURATIONS environment variable exists AND is equal to "true", then...
+    if [[ "${BUILD_ALL_CONFIGURATIONS:-false}" = "true" ]]; then
+        echo "This is a nightly build. Generating the appropriate steps..."
+        
+        # Editor builds (Test and Shipping build states do not exist for the Editor build target)
+        for build_state in "DebugGame" "Development"; do
+             upload_build_configuration_step "$1" "Win64" "Editor" "$build_state"
+        done
+
+        # NoEditor, Client and Server builds
+        if [[ "$1" == *"4.22"* ]]; then
+            # Prebuilt engines of native 4.22 and prior do not support Client and Server targets.
+            # We use prebuilt engines in CI, but have manually added two Server configurations:
+            for build_state in "Development" "Shipping"; do
+                upload_build_configuration_step "$1" "Linux" "Server" "$build_state"
+            done
+
+            # NoEditor builds
+            for build_platform in "Win64" "Linux"; do
+                for build_state in "DebugGame" "Development" "Shipping" "Test"; do
+                    upload_build_configuration_step "$1" "$build_platform" "" "$build_state"
+                done
+            done
+        else {
+            # Generate all possible builds for non-Editor build targets
+            for build_platform in "Win64" "Linux"; do
+                for build_target in "" "Client" "Server"; do
```

Comment:
What does `build_target == ""` mean? What am I building for?

Reply:
It's the `[empty]` build target. It includes both `Client` and `Server` code. See https://docs.unrealengine.com/en-US/Programming/Development/BuildConfigurations/index.html